### PR TITLE
Optimize cluster availability; Support tenant upgrading; Enhance webhook

### PR DESCRIPTION
**Pull request by PowerfooI**
## Summary

1. Add annotation to separate PVC and observer pod to avoid pvc deletion when pod deleted
2. When deleting observer resource, if there is no units on deleting observer, mark the observer as deleted. Leave OB system handle the long tail deletion.
3. Support tenant upgrading after restoring.
4. Enhance the webhook validating of OBCluster, OBTenant and OBTenantOperation

### CODE REVIEW

File: pkg/resource/obtenantoperation_task.go [hunk 1]
```diff
@@ -313,3 +314,76 @@ func (m *ObTenantOperationManager) retryUpdateTenant(obj *v1alpha1.OBTenant) err
...
+		}
+	}
+
+	if sysCompatible >= "4.1.0.0" && restoredCompatible < sysCompatible {
+		err := con.UpgradeTenantWithName(targetTenant.Spec.TenantName)
+		if err != nil {
+			return err
+		}
+	outer:
+		for i := 0; i < oceanbaseconst.DefaultStateWaitTimeout/5+1; i++ {
```
[the remark this answers is not in the document]
Comment (PowerfooI) [reply pkg/resource/obtenantoperation_task.go]: In order to decrease times operator query the DB, 5-sec interval is reasonable. I will declare a variable ahead.

File: pkg/resource/obcluster_task.go [hunk 2]
```diff
@@ -196,6 +196,7 @@ func (m *OBClusterManager) CreateOBZone() error {
 		BlockOwnerDeletion: &blockOwnerDeletion,
 	}
 	ownerReferenceList = append(ownerReferenceList, ownerReference)
+	sepVolumeAnnoVal, sepVolumeAnnoExist := GetAnnotationField(m.OBCluster, oceanbaseconst.AnnotationsIndependentPVCLifecycle)
```
[the remark this answers is not in the document]
Comment (PowerfooI) [reply pkg/resource/obcluster_task.go]: OK, I've resolved it.

**Comment by PowerfooI** [comment]
fix #115 by adding an annotation `oceanbase.oceanbase.com/independent-pvc-lifecycle` to obcluster, which denotes that ob-operator should not set the observer pod as ownerReference of creating PVC.

**Comment by PowerfooI** [comment]
enhance webhook validations of OBCluster and OBTenant #118

**Comment by PowerfooI** [comment]
implement tenant upgrade with OBTenantOperation #116

**Comment by PowerfooI** [comment]
Decrease waiting time of OBServer resource when being deleting by checking units transfer situations. #117

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Pull request merged**